spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/AbstractPaginatedResponse.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+abstract class AbstractPaginatedResponse {
+
+	private volatile Integer totalResults;
```

Comment:
I was encouraged to use `volatile` when other pieces of hidden code update/read the field. In general it does no harm to assume that for any field.  In my opinion this oughtn't to be necessary for spring/jackson accessed fields as there ought to be correct synchronisation observed in those implementations. But just in case, Ben Hale got me to put those in there.